test/core.jl

```diff
@@ -5089,3 +5089,23 @@ m22929_2.x = m22929_1
 @test !isdefined_22929_x(m22929_1)
 @test isdefined_22929_1(m22929_2)
 @test isdefined_22929_x(m22929_2)
+
+# issue 18933
+module GlobalDef18933
+    using Base.Test
+    f() = (global sin; nothing)
```

Comment:
Hm, this is kind of a weird case. Normally `global sin` at the toplevel would create a binding in the current module. But I see that since `f` doesn't assign to `sin`, maybe it shouldn't here. I guess this behavior is probably ok.